### PR TITLE
Initial commit for FileOperator.h and FileOperator.cpp with unit test

### PR DESCRIPTION
**Pull request by LinRenHong**

### CODE REVIEW

**Review by Jazz86 — APPROVED** [review]
LGTM.

File: src/Implement/FileOperator.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    else
+        return 500;
+}
+
+int FileOperator::Download(const std::unique_ptr<OperateData> &data) {
+    // where to download include fileName
+    std::string path = data->body["path"];
+    
+    // chunk number
+    unsigned long pos = std::stoi(data->body["pos"]);
```
Comment (john0312) [inline src/Implement/FileOperator.cpp]: Maybe add error handling for this?

File: src/Implement/FileOperator.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    std::string binary_data = "";
+    
+    if (DownloadStream(path, 2 * 1024 * 1024).ReadChunk(pos, &binary_data) == 1)
+        return 200;
+    else
+        return 500;
+}
+
+void FileOperator::CallingAPI(std::unique_ptr<OperateData> &data) {
+
```
Comment (john0312) [inline src/Implement/FileOperator.cpp]: Please clang-format to get rid of the extra lines.

File: src/Implement/FileOperator.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    
+    const int client_fd = std::stoi(data->header["client_file_descriptor"]);
```
Comment (john0312) [inline src/Implement/FileOperator.cpp]: Same as above on error handling.

File: src/Implement/UnitTestFileOperator.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    char buffer[2048] = {};
+    size_t n_recv = recv(sockfd, buffer, sizeof(buffer), 0);
+
+    temp_response = buffer;
+
+    close(sockfd);
+}
+
+// test with a sudo string data
+TEST(RESPONSEHANDLER_TEST, Basic) {
```
Comment (john0312) [inline src/Implement/UnitTestFileOperator.cpp]: Here you don't have the issue of not being able to mock a socket, so you can create a MockResponseHandler and expect the result in MockResponseHandler instead of going through the socket.

File: src/Implement/FileOperator.cpp (hunk 3 above)
Comment (john0312) [reply src/Implement/FileOperator.cpp]: Consider passing the fd in through the response handler for easier unit testing.

File: src/Implement/FileOperator.cpp (hunk 2 above)
Comment (john0312) [reply src/Implement/FileOperator.cpp]: This is not done yet.

File: src/Implement/UnitTestFileOperator.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        size_t n_recv = recv(sockfd, buffer, sizeof(buffer), 0);
+
+        temp_response = buffer;
+
+        close(sockfd);
+    }
+}
+
+// test with a sudo string data
+TEST(FILEOPERATOR_TEST, Basic) {
```
Comment (john0312) [inline src/Implement/UnitTestFileOperator.cpp:42]: As previously mentioned, consider adding MockResponseHandler and test the FileOperator class with it. You can do this in another PR.

**Review by john0312 — APPROVED** [review]
Approved assuming reviews are addressed.

**Review by AriaZhou — APPROVED** [review]

File: src/Header/FileOperator.h [hunk 6]
```diff
@@ -2,40 +2,39 @@
...
-  ResponseHandler response;
+// File management for client
+class FileOperator : public APIHandler
+{
+public:
+    // Constructor
+    FileOperator();
+    
+    // Upload file to server.
+    int Upload(const std::unique_ptr<OperateData> &data);
```
Comment (john0312) [inline src/Header/FileOperator.h]: Usually we pass std::unique_ptr in without the const or reference, because we're passing the ownership in with unique_ptr.

Please change this to without const and without reference.
Note that callers of this function should use std::move() to pass the unique_ptr in.

File: src/Implement/UnitTestFileOperator.cpp (hunk 5 above)
Comment (LinRenHong) [reply src/Implement/UnitTestFileOperator.cpp:42]: CallingAPI() itself will call ResponseHandler to send the result back to the client, so not need to add MockResponseHandler.
